app/javascript/controllers/spinner_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus";
+
+function preventEscapeListener(event) {
+  if (event.key === "Escape") {
+    event.preventDefault();
+    event.stopPropagation();
+  }
+}
+
+export default class extends Controller {
+  static targets = ['submit'];
+
+  submitStart() {
+    document.addEventListener("keydown", preventEscapeListener, true);
+    document.querySelector(".dialog--header").classList.add("opacity-20");
```

Comment:
**suggestion:** Instead of dimming all the elements of the dialog, why don't you try giving the background to the spinner that spans the dialog.  You could just put a blur on that and it might look pretty good and easier that adding and removing all those opacity.

Can you also add some text under the spinner that says something like "Transfering samples, this might take a while".

Reply:
I can, it was just a copy from a Flowbite example.